lib/install/management/store_files.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package management
+
+import (
+	"path"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25/types"
+	"github.com/vmware/vic/lib/metadata"
+	"github.com/vmware/vic/lib/portlayer/storage/vsphere"
+	"github.com/vmware/vic/pkg/errors"
+	"github.com/vmware/vic/pkg/vsphere/tasks"
+	"github.com/vmware/vic/pkg/vsphere/vm"
+
+	"golang.org/x/net/context"
+)
+
+const (
+	volumeRoot = "volumes"
+)
+
+func (d *Dispatcher) DeleteDataStores(vchVM *vm.VirtualMachine, conf *metadata.VirtualContainerHostConfigSpec) error {
+	ds, err := d.session.Finder.Datastore(d.ctx, conf.ImageStores[0].Host)
+	if err != nil {
+		err = errors.Errorf("Failed to find image datastore %s", conf.ImageStores[0].Host)
+		return err
+	}
+	d.session.Datastore = ds
+
+	path, err := d.getVCHRootDir(vchVM)
+	if err != nil {
+		return err
+	}
+
+	var errs []string
+	var emptyImages bool
+	var emptyVolumes bool
+	log.Infof("Removing images")
+	if emptyImages, err = d.deleteImages(ds, path); err != nil {
+		errs = append(errs, err.Error())
+	}
+	log.Infof("Removing volumes")
+	if emptyVolumes, err = d.deleteVolumes(ds, path); err != nil {
+		errs = append(errs, err.Error())
+	} else if !emptyVolumes {
+		log.Infof("Volumes directory %s is not empty, to delete with --force specified", path)
+	}
+
+	if emptyImages && emptyVolumes {
+		// if not empty, don't try to delete parent directory here
+		log.Infof("Removing root directory")
```

Comment:
`Removing stores directory` - let's not alarm users... the person running vic-machine and reading this may have a very different interpretation of that message that results in a reflective Ctrl-C to kill whatever says it.